util-eval/src/test/scala/com/twitter/util/EvalTest.scala

```diff
@@ -245,5 +245,12 @@ class EvalTest extends WordSpec {
         assert(eval.errors.isEmpty)
       }
     }
+
+    "eval with overriden classloader" in {
+      val eval = new Eval{
+        override def classLoader = this.getClass.getClassLoder
+      }
+      assert(eval.apply[Int]("1 + 1") == 2)
```

Comment:
TBH, I'm not super familiar with using different class loaders, but it doesn't appear to me that this test will exercise the logic you've implemented.
Maybe I'm wrong, so just let me know if that's the case.